.github/workflows/foreman.yml

```diff
@@ -0,0 +1,67 @@
+---
+name: Foreman
+
+on:
+  - pull_request
+
+env:
+  RAILS_ENV: test
+  DATABASE_URL: postgresql://postgres:password@localhost/test
+  DATABASE_CLEANER_ALLOW_REMOTE_DATABASE_URL: true
+  BUNDLE_WITHOUT: "console:development:journald"
+
+jobs:
+  setup_matrix:
+    name: Setup matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.build_matrix.outputs.matrix }}
+    steps:
+      - name: Build test matrix
+        id: build_matrix
+        uses: theforeman/gha-matrix-builder@v0
+  test:
+    name: "develop with Ruby ${{ matrix.ruby }} and Node ${{ matrix.node }}"
+    runs-on: ubuntu-latest
+    needs: setup_matrix
+    services:
+      postgres:
+        image: 'postgres:12'
+        ports: ['5432:5432']
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+        env:
+          POSTGRES_PASSWORD: password
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: ${{ fromJson(needs.setup_matrix.outputs.matrix).ruby }}
+        node: ${{ fromJson(needs.setup_matrix.outputs.matrix).node }}
+    steps:
+      - run: sudo apt-get update
+      - run: sudo apt-get install build-essential libcurl4-openssl-dev zlib1g-dev libpq-dev libvirt-dev
+      - name: "Check out Foreman develop"
+        uses: actions/checkout@v4
+      - name: "Set up Ruby ${{ matrix.ruby }}"
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+      - name: Archive Gemfile.lock
+        uses: actions/upload-artifact@v3
+        with:
+          name: Gemfile.lock
+          path: Gemfile.lock
+      - name: "Set up Node ${{ matrix.node }}"
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Prepare test env
+        run: |
+          bundle exec rake db:create
+          bundle exec rake db:migrate
```

Comment:
Just noticed that after copy-pasting...

Shouldn't we add here `db:seed` as well? Or do we auto-seed test DB as well? Maybe that's why libvirt test fails on GHA, but passes on Jenkins...

Reply:
tests should not seed but use fixtures, that's fine

(we should still call `db:seed` to ensure the seeds work, but ideally on an otherwise clean db, can be omitted in this context and added later)

Reply:
https://github.com/theforeman/foreman/pull/9933